src/components/SingleArticle.jsx

```diff
@@ -21,6 +22,11 @@ function SingleArticle() {
       });
   }, []);
 
+  function handleVote(vote) {
+    setVoteChange((currentVote) => currentVote + vote);
+    patchArticleVote(article_id, vote);
```

Comment:
Nice implementation of optimistic rendering, however we're missing one important element
If the patch request is unsuccessful (say for example the internet fails), it still looks to the user like everything is fine and the vote has gone through. We need to make sure there's some indication to the user if this request fails that their vote hasn't worked.
Here's the relevant section of the notes: https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#error-handling